File: 9-colecoes/src/Aluno.java

public class Aluno implements Comparable<Aluno> {

    int matricula;
    String nome;

    public Aluno(int m, String n) {
        this.matricula = m;
        this.nome = n;
    }

    @Override
    public boolean equals(Object o) {
        if (o instanceof Aluno) {
            Aluno e = (Aluno) o;
            return this.matricula == e.matricula;
        }
        return false;
    }

    @Override
    public int hashCode() {
        return ("" + this.matricula).hashCode();
    }

    @Override
    public int compareTo(Aluno aluno) {
        if (this.matricula < aluno.matricula) {
            return -1;
        }
        return 1;
    }

    @Override
    public String toString() {
        return "Aluno{" + "matricula=" + matricula + ", nome=" + nome + '}';
    }
    
    
}

File: 4-pesquisa/PesquisaEmColecoesOO/ArvoreHash/Aluno.cs
using System;
using System.Collections;
public class Aluno : IComparable{
    public int Matricula {get;set;}
    public string Nome {get;set;}

    public Aluno(int m, string n) {
        Matricula = m;
        Nome = n;
    }

    public override string ToString()
    {
        return "Nome: " + Nome + ". Matrícula: " + Matricula;
    }

    public override bool Equals(object? obj)
    {        
        if (obj.GetType() == GetType()) 
        {
            Aluno e = (Aluno) obj;
            return this.Matricula == e.Matricula;
        }
        return false; 
    }

    public int CompareTo(object obj) {
        Aluno a = (Aluno)obj;
        if (this.Matricula == a.Matricula) {
            return 0;
        } else if (this.Matricula < a.Matricula) {
            return -1;
        }
        return 1;
    }
}